### PR TITLE
7.0 carrier option - split state field in two boolean fields

### PR DESCRIPTION
**Pull request by bealdav**
with the same feature in base delivery_carrier_label

replace this PR https://github.com/OCA/carrier-delivery/pull/11

Sorry for delay

Thanks

### CODE REVIEW

**Comment by rvalyi** [comment]
:+1: LGTM

NOTE: flake8 tests are failing for style issues unrelated to this commit, so no problem merging that for me.

Now @bealdav, if you look at failing style tests here https://travis-ci.org/OCA/carrier-delivery/jobs/34522590#L157 it's mostly a few long lines to cut. That would be cool to fix them so we can get green builds which makes reviewing easier in the future.

File: base_delivery_carrier_label/stock.py [hunk 1]
```diff
@@ -185,8 +186,8 @@ def option_ids_change(self, cr, uid, ids, option_ids, carrier_id,
             return res
         carrier = carrier_obj.browse(cr, uid, carrier_id, context=context)
         for available_option in carrier.available_option_ids:
-            if (available_option.state == 'mandatory'
-                    and available_option.id not in option_ids[0][2]):
+            if (available_option.mandatory
+                    and not available_option.id in option_ids[0][2]):
```
Comment (guewen) [inline base_delivery_carrier_label/stock.py]: The pythonic form is `variable not in collection`, not `not variable in collection`

**Comment by guewen** [comment]
You can directly push on your branch to update your PR, no need to create a new one.

**Comment by bealdav** [comment]
@rvalyi flake8 limit char is 79 not 80.
I have updated my IDE column checker.
Maybe some OCA contributors are also confused like me on this point

**Comment by guewen** [comment]
@bealdav 79 chars is the default limit of all the linters (flake8, pep8, pyflakes, pylint) because this is the limit defined by the pep 8. When we speak about 80, it is 79 chars + '\n'.

**Comment by bealdav** [comment]
Is OK to merge ?
thanks

**Comment by guewen** [comment]
Thanks for the change 
:+1:

**Comment by gurneyalex** [comment]
needs rebasing

**Comment by bealdav** [comment]
║ test_flake8 FAIL
║ test_pylint Success
║ test_server Success
modules failed flake8 are not linked to this PR

**Comment by yvaucher** [comment]
PR #32 will solves flakes8 issue

About this one:

```
git rebase -i origin/7.0
for each conflict:
    hack hack
    git add .
    git rebase --continue
```

**Comment by sebastienbeau** [comment]
rebase done it's ok :+1:

**Comment by sylvainc** [comment]
:+1:

**Comment by yvaucher** [comment]
:+1:

**Comment by guewen** [comment]
This commit breaks (at least) the module `delivery_carrier_label_postlogistics`: https://github.com/OCA/carrier-delivery/issues/43. Some grepping to see if it doesn't break other modules would have been nice. It also shows that tests are missing.

**Comment by guewen** [comment]
Fixed in https://github.com/OCA/carrier-delivery/pull/44

**Pull request merged**